src/components/tl/mlx5/mcast/tl_mlx5_mcast_team.c

```diff
@@ -227,20 +212,77 @@ ucc_status_t ucc_tl_mlx5_mcast_coll_setup_comm_resources(ucc_tl_mlx5_mcast_coll_
     comm->pending_recv = 0;
     comm->buf_n        = comm->params.rx_depth * 2;
 
-    ret = posix_memalign((void**) &comm->pp_buf, page_size, buf_size * comm->buf_n);
-    if (ret) {
-        tl_error(comm->ctx->lib, "posix_memalign failed");
-        return UCC_ERR_NO_MEMORY;
-    }
+    if (comm->device_mem_enabled) {
+        /* TODO add check to make sure GPUDirect is enabled
+         * lsmod | grep nv_peer */
+        CUDA_FUNC(cudaMalloc((void **)&comm->grh_buf, GRH_LENGTH * sizeof(char)));
```

Comment:
why not using `ucc_mc_alloc`?

Reply:
Because, we would like to keep mcast lib as self-contained as possible, making it easier to port our UCC changes to other libraries with mcast support.